### PR TITLE
Add doc comments to model

### PR DESCRIPTION
**Pull request by dtassone**

### CODE REVIEW

File: packages/grid/x-grid-modules/src/gridComponent.tsx [hunk 1]
```diff
@@ -27,6 +27,10 @@ import { useApi, useColumns, useKeyboard, useRows } from './hooks/root';
 import { useLogger, useLoggerFactory } from './hooks/utils';
 import { debounce, mergeOptions } from './utils';
 
+/**
+ * Material-UI Grid React component implementing [[GridComponentProps]]
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/gridComponent.tsx:31]:
```suggestion
 * Material-UI Grid React component implementing [[GridComponentProps]].
```

File: packages/storybook/src/documentation/pages/styling.stories.mdx [hunk 2]
```diff
@@ -1,52 +1,51 @@
...
-<XGrid rows={rows} columns={columns} options={{ autoHeight: true }} />
+const columns: Columns = [
+  { field: 'styledCell', cellClass: 'super-app-theme--cell', align: 'center'},
+  {
+    field: 'withRules',
+    cellClassRules: {
+      negative: {value} => value < 0,
+      positive: {value} => value > 0
+    },
+    align: 'right'
```
Comment (oliviertassinari) [inline packages/storybook/src/documentation/pages/styling.stories.mdx:46]:
```suggestion
    align: 'right',
```

**Comment by dtassone** [comment]
> What about the leading dots at the end of the description of the props. Should we enforce this convention, considering them sentences? I would be in favor of it. cc @mbrookes

I don't really see it as a blocker for this PR, I would like to merge this one and the useApiRef one to update the doc about the changes with useApiRef. For the dots we can always do it in the next one, or add a prettier rule

**Comment by dtassone** [comment]
> > For the dots we can always do it in the next one
> 
> Which will be harder to fix if we delay it, we won't have the diff here
> 
> > prettier rule
> 
> Any idea on how to do that? It might be spending a lot of time for a contention we can enforce in the review steps, not sure it worth it

done it with regex ;)

File: packages/grid/x-grid-modules/src/models/params/cellParams.ts [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+   * A function that let you get data from other columns
+   * @param field
+   */
   getValue: (field: string) => CellValue;
+  /**
+   * The full set of data of the row that the current cell belongs to
+   */
   data: RowData;
+  /**
+   * The row model  of the row that the current cell belongs to
```
[the remark this answers is not in the document]
Comment (oliviertassinari) [reply packages/grid/x-grid-modules/src/models/params/cellParams.ts:21]: You forgot about this one :)

File: packages/grid/x-grid-modules/src/gridComponent.tsx [hunk 4]
```diff
@@ -27,6 +27,10 @@ import { useApi, useColumns, useKeyboard, useRows } from './hooks/root';
 import { useLogger, useLoggerFactory } from './hooks/utils';
 import { debounce, mergeOptions } from './utils';
 
+/**
+ * Material-UI Grid React component implementing [[GridComponentProps]].
+ * @return JSX.Element.
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/gridComponent.tsx:32]: I doubt the point here will be correctly interpreted.

**Comment by oliviertassinari** [comment]
> done it with regex ;)

Well, not everything was handled but it's a start :).

Regarding the philosophy, we should use, I think that it's really important that we aim for the boy scout rule: always leave a place in a better shape you entered it ✨.

**Comment by mbrookes** [comment]
> Should we enforce this convention, considering them sentences? I would be in favor of it. cc @mbrookes

Seems that you already arrived at a conclusion (👌), but yes, we should follow the same conventions for prop descriptions as used in core and lab. There are some common patterns to look out for, for example for boolean props, "If `true`, ...", amongst others.

**Pull request merged**